File: h-index1.py
# Time Complexity : O(NlogN)
# Space Complexity : O(1)
# Did this code successfully run on Leetcode : Yes

# Your code here along with comments explaining your approach
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        if citations == None or len(citations) == 0:
            return 0
        citations.sort()
        n = len(citations)
        for i in range(n):
            citation = citations[i]
            count = n-i
            
            if citation >= count:
                return count
            
        return 0
        

File: hIndex1.java
// Time Complexity : O(2n) ->O(n)
// Space Complexity : O(n)
// Did this code successfully run on Leetcode : Yes
public class hIndex1 {
        public int hIndex(int[] citations) {
            int n = citations.length;
            int[] bs = new int[n+1];

            for(int i=0; i<n; i++)
            {
                if(citations[i]>n)
                {
                    bs[n]++;
                }
                else
                {
                    bs[citations[i]]++;
                }
            }
            int sum = 0;
            for(int i = n; i>=0; i--)
            {
                sum += bs[i];
                if(i <= sum)
                {
                    return i;
                }
            }

            return -1;
        }
    }
